src/DynamoCoreWpf/Views/CodeBlocks/CodeBlockEditor.cs

```diff
@@ -64,6 +109,35 @@ protected override void OnCommitChange()
                 CommitChanges(recorder);
 
             createdForNewCodeBlock = false; // First commit is now over.
+
+            stateMachine.Transit(EditorStateMachine.EditorState.Commited);
+        }
+
+        protected override void OnEditorTextChanged()
+        {
+            stateMachine.Transit(EditorStateMachine.EditorState.Editing);
+        }
+
+        private void OnEditorStateChanged(EditorStateMachine.EditorState state)
+        {
+            switch (state)
+            {
+                case EditorStateMachine.EditorState.Editing:
+                    EnableOutputPorts(false); 
+                    break;
+
+                case EditorStateMachine.EditorState.Commited:
+                    EnableOutputPorts(true); 
+                    break;
+            }
+        }
+
+        private void EnableOutputPorts(bool isEnabled)
+        {
+            for (int i = 0; i < codeBlockNode.OutPorts.Count; i++)
```

Comment:
If we end up removing some variables during CBN edit, wouldn't that affect input ports also?